Core/HLE/sceNetInet.cpp

```diff
@@ -0,0 +1,906 @@
+// Copyright (c) 2012- PPSSPP Project.
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 2.0 or later versions.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#if __linux__ || __APPLE__ || defined(__OpenBSD__)
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#endif
+
+#include "Common/Net/Resolve.h"
+#include "Common/Data/Text/Parsers.h"
+
+#include "Common/Serialize/Serializer.h"
+#include "Common/Serialize/SerializeFuncs.h"
+#include "Core/Config.h"
+#include "Core/HLE/HLE.h"
+#include "Core/HLE/FunctionWrappers.h"
+#include "Core/MIPS/MIPS.h"
+#include "Core/MemMapHelpers.h"
+
+#include "Core/HLE/proAdhoc.h"
+#include "Core/HLE/sceNet.h"
+#include "Core/HLE/sceNetInet.h"
+
+#include <iostream>
+#include <shared_mutex>
+
+#include "Core/HLE/sceNp.h"
+#include "Core/Reporting.h"
+// TODO: move Core/Net
+#include "Core/Net/InetCommon.h"
+#include "Core/Net/SceSocket.h"
+
+#if PPSSPP_PLATFORM(SWITCH) && !defined(INADDR_NONE)
+// Missing toolchain define
+#define INADDR_NONE 0xFFFFFFFF
+#elif PPSSPP_PLATFORM(WINDOWS)
+#pragma comment(lib, "ws2_32.lib")
+#define close closesocket
+#define ERROR_WHEN_NONBLOCKING_CALL_OCCURS WSAEWOULDBLOCK
+using netBufferType = char;
```

Comment:
TODO: document why this is necessary, C++ reinterpret_cast dealing with differing C headers between winsock and sane POSIX sockets implementations